integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -567,28 +573,50 @@ private int writeInternal(String path, ByteBuffer buf, long size, long offset, F
         }
       }
     }
-
-    FileOutStream os = ce.getOut();
-    long bytesWritten = os.getBytesWritten();
-    if (offset != bytesWritten && offset + sz > bytesWritten) {
-      LOG.error("Only sequential write is supported. Cannot write bytes of size {} to offset {} "
-          + "when {} bytes have written to path {}", size, offset, bytesWritten, path);
-      return -ErrorCodes.EIO();
-    }
-    if (offset + sz <= bytesWritten) {
-      LOG.warn("Skip writting to file {} offset={} size={} when {} bytes has written to file",
-          path, offset, sz, bytesWritten);
-      // To fulfill vim :wq
-      return sz;
-    }
-
-    try {
-      final byte[] dest = new byte[sz];
-      buf.get(dest, 0, sz);
-      os.write(dest);
-    } catch (IOException e) {
-      LOG.error("IOException while writing to {}.", path, e);
-      return -ErrorCodes.EIO();
+    if (!mTmpFolder.isEmpty()) {
+      RandomAccessFile tmpFile;
+      try {
+        File tmpFolder = new File(mTmpFolder);
+        if (!tmpFolder.exists()) {
+          tmpFolder.mkdirs();
+        }
+        tmpFile = new RandomAccessFile(Paths.get(mTmpFolder, path).toString(), "rw");
+      } catch (FileNotFoundException e) {
+        LOG.error("Failed to create temporary file {}: ", Paths.get(mTmpFolder, path), e);
+        return -ErrorCodes.EIO();
+      }
+      try {
+        final byte[] dest = new byte[sz];
+        buf.get(dest, 0, sz);
+        tmpFile.seek(offset);
+        tmpFile.write(dest);
+        tmpFile.close();
```

Comment:
if seek or write throws IOE, this tmpFile will not be closed.
that's why we shall use try-with-resource.